### DISCUSSION TITLE
getting EXC_BAD_ACCESS while calling overture methods in static vars

### DISCUSSION THREAD

**Discussion by jonchui** [pointfreeco/swift-overture#66] [Q&A]
here are some examples of EXC_BAD_ACCESS crashes i'm getting while running my unit tests locally

![image](https://user-images.githubusercontent.com/204868/61749093-2245aa00-ad5f-11e9-9eb0-73fd741e83d6.png)

![image](https://user-images.githubusercontent.com/204868/61750567-fd533600-ad62-11e9-88d1-c3c76693cea5.png)

I fixed the later one by changing:
```
static var burningDesirePostItem = with(BurningDesireFeedCellItem(), concat(
        mut(\.post, .mockRegular)
    ))
```

to 

```
    static var burningDesirePostItem : BurningDesireFeedCellItem = {
        let burningDesireItem = BurningDesireFeedCellItem()
        burningDesireItem.post = SGPost.mockRegular
        return burningDesireItem
    }()
```

But I'm not sure if i even fully understand what the problem is? I'm assuming that in the static instances, something hasn't been setup for overture, which is expecting to be called in instance methods?

The former one i can't seem to figure out a good way to fix - it seems the last line in:

```
static let mock = {
        return with(
            CoachingAPI(),
            concat(
                mut(\.selectedCoachId, { return .init(String(Coach.mockCoaches.first!.id)) }),
                mut(\.subscriptionStatus, { return .enabled }),
                mut(\.fetchCoaches, { $0(.success(Coach.mockCoaches)) })
            )
        )
    }()
```

Is it effectively mocking out the return of `fetchCoaches` function on `CoachingAPI`?

**Comment by jonchui** [comment]
this is only happening on local dev , NOT on bitrise, weirdly enough

**Comment by mbrandonw** [comment]
Hi @jonchui! Sorry for the delay in responding!

We've occasionally encountered this kind of thing before, and it usually requires fixing the key paths to be more explicit by avoiding shorthand syntax.

We're also curious what kinds of values `CoachingAPI` and `BurningDesireFeedCellItem` are? Are they classes or structs?

**Comment by stephencelis** [comment]
Going to migrate to a discussion in case folks encounter this in the future.
